### PR TITLE
Add lines for indentation

### PR DESCRIPTION
**Pull request by fanwgwg**
[*DYN-869 In tree view of Library, we should have the connecting Lines for drop down arrows.*](https://jira.autodesk.com/browse/DYN-869)

capture2

### Reviewer
@Benglin 

### FYI
@riteshchandawar

### CODE REVIEW

**Comment by Racel** [comment]
Nice @fanwgwg!. Can you make the arrows a bit smaller?

**Comment by fanwgwg** [comment]
@Racel I've updated the arrow size. Does this size look okay?

capture2

**Comment by Racel** [comment]
Awesome! looks good @fanwgwg

**Comment by fanwgwg** [comment]
@ellensi Can you help review this PR? Thanks!

File: src/resources/LibraryStyles.css [hunk 1]
```diff
@@ -154,13 +153,18 @@ input {
...
+
+.BodyIndentationWithLine,
+.BodyIndentationWithoutLine {
+    margin-bottom: 18px;
+    padding-left: 18px;
 }
 
-.BodyIndentation {
-    padding: 8px;
+.BodyIndentationWithLine {
```
Comment (ellensi) [inline src/resources/LibraryStyles.css]: Hi @fanwgwg,

Newly introduced elements such as `.BodyIndentationWithLine` and `.BodyIndentationWithoutLine` are actually just margins and do not need to be separate elements.

We can assign multiple class names in React:
```ts
className={LibraryItemBodyContainer}                     /* BodyIndentationWithoutLine */
className={LibraryItemBodyContainer HasExpanderLine}        /* BodyIndentationWithLine */
```

```css
.HasExpanderLine {
    margin-left: 18px;
}

.HasExpanderLine::before {
    content: "";
    border-right: 1px solid #808080;
    margin-bottom: 18px;
}
```

**Comment by riteshchandawar** [comment]
@Racel In old UI we were having some gap between arrows and lines, I think that looks better than the one which we implemented above. What do you say?

image

**Comment by Racel** [comment]
@riteshchandawar - I agree with you, the slight space makes it a bit clearer that the arrows are also buttons. Can we try it out and see?

File: src/components/LibraryItem.tsx [hunk 2]
```diff
@@ -139,17 +139,31 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
         let header: JSX.Element = null;
```
Comment (Benglin) [inline src/components/LibraryItem.tsx]: The `render` method is going out of control, can you refactor it into multiple smaller render methods?

```js
render() {
    if (!this.props.data.visible) {
        return null;
    }

    let iconElement: JSX.Element = renderItemIcon();

    let nestedElements: JSX.Element = null;
    let clusteredElements: JSX.Element = null;
    if (this.state.expanded && this.props.data.childItems.length > 0) {
        // Same stuff goes here...
    }

    let header: JSX.Element = renderItemHeader();
    let bodyIndentation: JSX.Element = renderBodyIndentation();
    let arrow: JSX.Element = renderItemArrow();

    return (existing stuff...);
}
```

**Comment by fanwgwg** [comment]
@Racel @riteshchandawar @ellensi  

I've increased the space. Does this look fine?

capture3

**Comment by Racel** [comment]
@fanwgwg - space looks good, but arrow size looks maybe a pixel or 2 too small. Can you increase it a little?

**Comment by fanwgwg** [comment]
@Racel  I've increased the arrow size a little bit. Is this size okay?

capture3

File: src/resources/LibraryStyles.css (hunk 1 above)
Comment (fanwgwg) [inline src/resources/LibraryStyles.css]: I've removed the body indentation line, but it's good to know this. Thanks!

**Comment by Racel** [comment]
Looks good @fanwgwg

**Pull request merged**